cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -313,22 +316,34 @@ private String addExecutable(String exec, List<String> args) {
   private void performLogOnError(ProcessResult result, int exitCode, String interpreter) {
 
     if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
-      String message = createCommandMessage(interpreter, " failed with exit code " + exitCode + "!");
+      IdeSubLogger subLogger = this.context.error();
+      IdeLogLevel ideLogLevel;
+      String message = createCommandMessage(interpreter, "\nfailed with exit code " + exitCode + "!");
+
       if (this.errorHandling == ProcessErrorHandling.THROW_CLI) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new CliProcessException(message, result);
       } else if (this.errorHandling == ProcessErrorHandling.THROW_ERR) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new IllegalStateException(message);
       }
```

Comment:
Why not just moving this code after the logging code below?
Than we have two clear aspects:
1. logging of errors with log-level according to error handling
2. throwing exception according to error handling

Reply:
Move throwing of exceptions after logging.